### PR TITLE
changed getEvents rpc, added 2 integrational tests

### PR DESCRIPTION
**Pull request by nikania**
# Pull Request type

- Other (please describe): enchancing

## What is the current behavior?

Resolves: #510

## What is the new behavior?

- changed the mechanism of getting events for the rpc endpoint `getEvents` by taking events from transaction receipts.
- removed `transaction_hash` field in `EventWrapper` structure
- `StarknetEvent(StarknetEventType)` also changed because `StarknetEventType` is an alias for the `EventWrapper` structure

## Other information

Maybe it's better to change pagination mechanism - for example - changing the `continuation_token` to tuple `continuation_token = (trx_receipt_index, event_index)` - or maybe even `continuation_token = (block_index, trx_receipt_index, event_index)` to make skipping already received events easier

### CODE REVIEW

**Review by EvolveArt — APPROVED** [review]
lgtm

**Comment by EvolveArt** [comment]
> Regarding your suggestion: Whatever change seems relevant as long as the rpc specs are not broken are more than welcome

Can't seem to find a precise spec for the continuation token, do you?

**Comment by nikania** [comment]
> > Regarding your suggestion: Whatever change seems relevant as long as the rpc specs are not broken are more than welcome
> 
> Can't seem to find a precise spec for the continuation token, do you?

I think this 
["type": "string"](https://github.com/starkware-libs/starknet-specs/blob/aaea417f193bbec87b59455128d4b09a06876c28/api/starknet_api_openrpc.json#L760)

File: crates/client/rpc/src/events/mod.rs [hunk 1]
```diff
@@ -99,40 +97,54 @@ where
...
-                    })
-                    .collect::<Vec<EmittedEvent>>(),
-            );
-
-            if filtered_events.len() >= chunk_size as usize {
-                let token = if index - index_before_loop < block_events_len { Some((index).to_string()) } else { None };
-                return Ok(EventsPage { events: filtered_events, continuation_token: token });
+            for receipt in trx_receipts.iter() {
+                let receipt_events_len: usize = receipt.events.len();
+                // skip receit with event if receipt.events.len() < continuation_token
```
[the remark this answers is not in the document]
Comment (nikania) [reply crates/client/rpc/src/events/mod.rs]: changed continuation_token therefore deleted this line

**Review by LucasLvy — APPROVED** [review]

**Comment by EvolveArt** [comment]
> > > Regarding your suggestion: Whatever change seems relevant as long as the rpc specs are not broken are more than welcome
> > 
> > 
> > Can't seem to find a precise spec for the continuation token, do you?
> 
> I think this ["type": "string"](https://github.com/starkware-libs/starknet-specs/blob/aaea417f193bbec87b59455128d4b09a06876c28/api/starknet_api_openrpc.json#L760)

I mean how should it look like as a `string` ?

**Comment by nikania** [comment]
Should I do something else or PR can be merged?

**Comment by nikania** [comment]
@LucasLvy lint fixed

**Pull request merged**